# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/InsertTextTransaction.cpp
# Commit: 1466d7d61102
# Full Hash: 1466d7d6110279cf167ab8b1567be693df5d7c3c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-03-03 09:50:30
# Regressor Bug: 1618089
# File Overlap Count: 2
# Description:
#   Bug 1618089 - part 7: Make some related methods use `EditorDOMPointInText` for arguments or result r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D64338
# ==============================================================================

diff -r 079f4b05ead2 -r 1466d7d61102 editor/libeditor/InsertTextTransaction.cpp
--- a/editor/libeditor/InsertTextTransaction.cpp	Tue Mar 03 02:19:59 2020 +0000
+++ b/editor/libeditor/InsertTextTransaction.cpp	Tue Mar 03 01:15:38 2020 +0000
@@ -20,18 +20,19 @@
 
 // static
 already_AddRefed<InsertTextTransaction> InsertTextTransaction::Create(
-    EditorBase& aEditorBase, const nsAString& aStringToInsert, Text& aTextNode,
-    uint32_t aOffset) {
-  RefPtr<InsertTextTransaction> transaction = new InsertTextTransaction(
-      aEditorBase, aStringToInsert, aTextNode, aOffset);
+    EditorBase& aEditorBase, const nsAString& aStringToInsert,
+    const EditorDOMPointInText& aPointToInsert) {
+  MOZ_ASSERT(aPointToInsert.IsSetAndValid());
+  RefPtr<InsertTextTransaction> transaction =
+      new InsertTextTransaction(aEditorBase, aStringToInsert, aPointToInsert);
   return transaction.forget();
 }
 
-InsertTextTransaction::InsertTextTransaction(EditorBase& aEditorBase,
-                                             const nsAString& aStringToInsert,
-                                             Text& aTextNode, uint32_t aOffset)
-    : mTextNode(&aTextNode),
-      mOffset(aOffset),
+InsertTextTransaction::InsertTextTransaction(
+    EditorBase& aEditorBase, const nsAString& aStringToInsert,
+    const EditorDOMPointInText& aPointToInsert)
+    : mTextNode(aPointToInsert.ContainerAsText()),
+      mOffset(aPointToInsert.Offset()),
       mStringToInsert(aStringToInsert),
       mEditorBase(&aEditorBase) {}
 